packages/strongbox/lib/index.ts

```diff
@@ -0,0 +1,282 @@
+ */
+export type Value = string | Buffer;
+
+/**
+ * An object representing a persisted item containing something of type `T` (which can be a {@linkcode Buffer} or an encoded string; see {@linkcode ItemEncoding}).
+ *
+ * A {@linkcode Item} does not know anything about where it is stored, or how it is stored.
+ * @typeParam T - Type of data stored in the item
+ */
+export interface Item<T extends Value> {
+  /**
+   * Encoding of underlying value
+   */
+  encoding: ItemEncoding;
+  /**
+   * Slugified name
+   */
+  id: string;
+  /**
+   * Name of item
+   */
+  name: string;
+
+  /**
+   * Reads value
+   */
+  read(): Promise<T | undefined>;
+  /**
+   * Writes value
+   * @param value New value
+   */
+  write(value: T): Promise<void>;
+
+  /**
+   * Deletes the item.
+   */
+  clear(): Promise<void>;
+
+  /**
+   * Last known value (stored in memory)
+   *
+   * @remarks A custom {@linkcode Item} meant to handle very large files should probably not implement this.
+   */
+  value?: T | undefined;
+}
+
+/**
+ * Set of known `Item` encodings
+ * @internal
+ */
+const ITEM_ENCODINGS: Readonly<Set<ItemEncoding>> = new Set([
+  'ascii',
+  'utf8',
+  'utf-8',
+  'utf16le',
+  'ucs2',
+  'ucs-2',
+  'base64',
+  'base64url',
+  'latin1',
+  'binary',
+  'hex',
+  null,
+]);
+
+/**
+ * Type guard for encodings
+ * @param value any
+ * @returns `true` is `value` is a valid file encoding
+ */
+function isEncoding(value: any): value is ItemEncoding {
+  return ITEM_ENCODINGS.has(value);
+}
+
+/**
+ * @see {@linkcode StrongboxOpts}
+ */
+export const DEFAULT_SUFFIX = 'strongbox';
+
+/**
+ * A constructor function which instantiates a {@linkcode Item}.
+ */
+export type ItemCtor<T extends Value> = new (
+  name: string,
+  container: string,
+  encoding?: ItemEncoding
+) => Item<T>;
+
+/**
+ * Main entry point for use of this module
+ *
+ * Manages multiple {@linkcode Item}s.
+ */
+export class Strongbox {
+  /**
+   * Default {@linkcode ItemCtor} to use when creating new {@linkcode Item}s
+   */
+  protected defaultCtor: ItemCtor<any>;
+  /**
+   * Slugified name of this instance; corresponds to the directory name.
+   *
+   * If `dir` is provided, this value is unused.
+   * If `suffix` is provided, then this will be the parent directory of `suffix`.
+   */
+  public readonly id: string;
+
+  /**
+   * Override the directory of this container.
+   *
+   * If this is present, both `suffix` and `containerId` are unused.
+   */
+  public readonly container: string;
+
+  /**
+   * Store of known {@linkcode Item}s
+   * @internal
+   */
+  protected items: Map<string, WeakRef<Item<any>>>;
+
+  /**
+   * Slugifies the name & determines the directory
+   * @param name Name of instance
+   * @param opts Options
+   */
+  protected constructor(
+    public readonly name: string,
+    {container, suffix = DEFAULT_SUFFIX, defaultCtor = BaseItem}: StrongboxOpts = {}
+  ) {
+    this.id = slugify(name);
+    this.defaultCtor = defaultCtor;
```

Comment:
maybe too wordy but `defaultItemCtor` might be more clear than `defaultCtor` (which could be taken to refer to the Strongbox constructor?)

Reply:
mm, yes.